src/lib/openapi-to-json-schema.ts

```diff
@@ -403,12 +437,17 @@ export default function toJSONSchema(
     } else if (schema.type === 'object') {
       if ('properties' in schema) {
         Object.keys(schema.properties).map(prop => {
-          schema.properties[prop] = toJSONSchema(schema.properties[prop] as RMOAS.SchemaObject, {
-            currentLocation: `${currentLocation}/${encodePointer(prop)}`,
-            globalDefaults,
-            prevSchemas,
-            refLogger,
-          });
+          if (
+            Array.isArray(schema.properties[prop]) ||
+            (typeof schema.properties[prop] === 'object' && schema.properties[prop] !== null)
+          ) {
+            schema.properties[prop] = toJSONSchema(schema.properties[prop] as RMOAS.SchemaObject, {
+              currentLocation: `${currentLocation}/${encodePointer(prop)}`,
+              globalDefaults,
+              prevSchemas,
+              refLogger,
+            });
+          }
```

Comment:
Made this change because I discovered that if there was an `x-` extension, and it was a primitive, in an object we'd run it through the generator again and completely the value.

![screen_shot_2022-01-10_at_3 44 13_pm](https://user-images.githubusercontent.com/33762/149013449-92b376eb-0de5-4e17-8262-07c07d0aedff.png)

Though thinking about it, maybe this should instead just exclude anything that starts with `x-`?